core/storage/audit/gae_models.py

```diff
@@ -79,3 +79,50 @@ def get_user_id_migration_policy():
     def get_user_id_migration_field(cls):
         """Return field that contains user ID."""
         return cls.user_id
+
+
+class UsernameChangeAuditModel(base_models.BaseModel):
+    """Records the changes made to usernames via the admin panel.
+
+    Instances of this class are keyed by a custom Id.
+    [user_id].[timestamp_in_sec]
+    """
+
+    # The user_id of the user that is making the change.
```

Comment:
For extra clarity, here and elsewhere, add: "(Note that this is typically an admin user, who would be a different user from the one whose username is being changed.)"

Reply:
Done.